app/Main.hs

```diff
@@ -251,6 +251,43 @@ onlyValentPoints =
               singlePath)
     []
 
+abc :: [[Records.Icon]] -> [[Records.Icon]]
+abc inputPaths =
+  foldl
+    (\acc1 rowIndex ->
+       (foldl
+          (\acc2 columnIndex ->
+             let inputPath = inputPaths !! columnIndex
+                 newPath =
+                   case acc1 of
+                     [] -> []
+                     otherwise -> acc1 !! columnIndex
+                 newIcon =
+                   if rowIndex >= length inputPath
+                     then Records.tempValentPoint ":x:"
+                     else inputPath !! rowIndex
+              in acc2 ++ [newPath ++ [newIcon]])
+          ([] :: [[Records.Icon]])
+          [0 .. ((length inputPaths) - 1)]))
+    ([] :: [[Records.Icon]])
+    [0 .. 12]
+
```

Comment:
maxPathLength suggestion is great, but ultimately this is now how the number of iterations will be determined - we could have higher number of rows depending on how many valent points need to be inserted/ I'll leave in for now as it will have to be replaced anyway in short term future